gcc/d/d-objfile.cc

```diff
@@ -533,31 +533,31 @@ ClassDeclaration::toObjFile()
   d_finish_symbol (csym);
 
   // Put out the vtbl[]
-  dt = NULL_TREE;
+  vec<constructor_elt, va_gc> *elms = NULL;
 
   // first entry is ClassInfo reference
   if (vtblOffset())
-    dt_cons (&dt, build_address (csym));
+    CONSTRUCTOR_APPEND_ELT (elms, size_zero_node, build_address (csym));
 
   for (size_t i = vtblOffset(); i < vtbl.dim; i++)
     {
       FuncDeclaration *fd = vtbl[i]->isFuncDeclaration();
 
-      if (fd && (fd->fbody || !isAbstract()))
-	{
-	  fd->functionSemantic();
+      if (!fd || (!fd->fbody && isAbstract ()))
+	continue;
```

Comment:
AFAICS this case corresponds to this old implementation code:
```c++
      else
	dt_cons (&dt, null_pointer_node);
```

Is it correct that we do not add any entry to the constructor in the new code?